src/_zkapauthorizer/model.py

```diff
@@ -816,6 +817,20 @@ def get_events(self, cursor):
 
         return EventStream(changes=tuple(Change(seq, stmt) for seq, stmt in rows))
 
+    @with_cursor
```

Comment:
Thinking further on this, all of the `event-stream`-interacting APIs form their own interesting little unit of functionality that's distinct from everything else `VoucherStore` does.  Perhaps these should be free functions that operate on an unreplicated connection.  They might even belong in `replicate.py`?

This also makes it possible to remove the `VoucherStore` attribute of `_ReplicationService` - which is the only reason `replicate.py` knows anything about `model.py`.

Reply:
i guess I was working on the assumption that `VoucherStore` was "the" database-interface ...

Reply:
For a while it was.  When lease maintenance was introduced that weakened a little bit (see `LeaseMaintenance` in model.py).  I think replication weakens this further.  What makes sense to me is for `model.py` (mainly `VoucherStore` and `LeaseMainteance`) to be the *replicated* database interface and for there to be a separate unreplicated database interface, with the `event-stream` stuff all belonging to the latter.

It would have been fine with me if we had ended up with just one database interface but with the replication design the consequence of that right now would be that accidentally calling `VoucherStore.add_event` (not `VoucherStore.add_event.wrapped` with your own unreplicated cursor) or `prune_events_to` means `event-stream` rows will be replicated - not what we want.  When I wrote my earlier comment I also thought this could lead to an infinite replication loop but now I realize that because of the importance / size threshold this is probably not very likely in practice (maybe not strictly impossible though).